### PR TITLE
[Fix-3840][server] Fix When the tenant code is a number, it should not be saved successfully

### PR DESCRIPTION
**Pull request by hermeshephaestus**
header
[DS-ISSUE3840][fix] fix ui

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/RegexUtils.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+package org.apache.dolphinscheduler.api.utils;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This is Regex expression utils.
+ */
+public class RegexUtils {
```
Comment (CalvinKirs) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/RegexUtils.java:26]: add a private constructor to hide the implicit public one

Comment (CalvinKirs) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/RegexUtils.java:26]: After changing this, there is no problem with this PR.

**Review by CalvinKirs — APPROVED** [review]
PTAL @xingchun-chen

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/RegexUtils.java (hunk 1 above)
Comment (hermeshephaestus) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/RegexUtils.java:26]: OK, thank you for your suggestion, I have added a private constructor, please check that!

**Comment by hermeshephaestus** [comment]
By the way, because the change of issue #3540 is really small, i have submitted the same pr together, please check the commits. @CalvinKirs @yangyichao-mango @yangxifi

**Comment by CalvinKirs** [comment]
> By the way, because the change of issue #3540 is really small, i have submitted the same pr together, please check the commits. @CalvinKirs @yangyichao-mango @yangxifi

looks good to me,Let's wait for xingchun-chen review

**Review by qiaozhanwei — APPROVED** [review]
+1

**Comment by CalvinKirs** [comment]
considering that it's your first contribution, I think we can get deep communiction, you can contact me through mail or add wechat(Kris_Evil), when mail or added, please tell me who you are, I think I can help to familiar with the DolphinScheduler if you meet with problems.

**Pull request merged**